storage/remote/queue_manager.go

```diff
@@ -398,6 +443,37 @@ func (t *QueueManager) Stop() {
 	t.metrics.unregister()
 }
 
+// Getting the data from a shard and structuring it as a SegmentRecord from a shard
+func (t *QueueManager) getSegmentRecords() SegmentRecord {
+	RecordSegment := t.shards.qm.watcher.SegmentFile
```

Comment:
`t` is already the queue manager, so this can be simplified to just `t.watcher.SegmentFile`.

Also, what would happen if you try to get the `SegmentFile` from the watcher at the same time the watcher was changing `SegmentFile`?

Reply:
Would this still be able to get each individual shards segment number?

Reply:
It's not clear what you're trying to do here actually. Shards, and also the queue manager, have no notion of the WAL or segment files.

Reply:
Ohh is this the watcher that's monitoring the segment file that WAL is writing on, not the one its getting read from?

Reply:
No, this is the watcher that reads the segment file and passes the samples back to the queue manager. I think the name of the function and your comments about shards are what's confusing me.

As Chris said, we can just access `t.watcher.SegmentFile`. My point was that shards and the queue manager have no notion or concept of WAL segments. So this question `Would this still be able to get each individual shards segment number?`, the shards do not have an individual segment number. The watcher is reading a segment, passing samples back to queue manager, and queue manager is deciding which of the shards to assign a given sample to.